docs/core/common/assets/js/api/core/hooks.md

```diff
@@ -0,0 +1,182 @@
+## API - `$e.hooks`
+*  **Description**: `$e.hooks` api is a manager of `$e.hooks.ui` & `$e.hooks.data`, allow you to create custom hooks.
+the hooks attached to $e.commands and each hook fired _after/before_ running a command, that runs by $e.run().
+*  **Location**: *core/common/assets/js/api/core/hooks.js*
+*  **Methods**:
+
+	| Method                               | Params                                                                                          |  Returns              | Description                                              
+	|--------------------------------------|-------------------------------------------------------------------------------------------------|-----------------------|-----------------------------------------------------------
+	| `$e.hooks.activate()`                |                                                                                                 |                       | Activate all hooks. 
+	| `$e.hooks.deactivate()`              |                                                                                                 |                       | Deactivate all hooks. 
+	| `$e.hooks.getAll()`                  |                                                                                                 | `{Array}`             | Receive all loaded hooks. 
+	| `$e.hooks.register()`                | `{String}` *type*, `{String}` *event*, `{HookBase}` *instance*                                  | `{Object}` *callback* | Register hook. 
+	| `$e.hooks.run()`                     | `{String}` *type*, `{String}` *event*, `{String}` *command*, `{Object}` *args*, `{*}` *result*  | `{Boolean}`           | Run's a hook. 
```

Comment:
> Run's a hook

You can write either "Run" or "Runs" (depending on your meaning), but not "Run's".

Reply:
Handled.